src/cpp/StatisticsBackend.cpp

```diff
@@ -634,6 +654,15 @@ void StatisticsBackend::load_database(
     details::StatisticsBackendData::get_instance()->database_->load_database(dump);
 }
 
+void StatisticsBackend::reset()
+{
+    if (!details::StatisticsBackendData::get_instance()->monitors_by_entity_.empty())
+    {
+        throw PreconditionNotMet("There are active monitors");
+    }
```

Comment:
TL;DR: `monitors_by_entity` is just a place where to hold the monitors that still hold active structures, i.e., active monitors.

The thing is that the monitor contains all the DDS entities to receive the statistics data, and those entities must be teared off when the monitor stops, making the monitor object empty and unnecessary anymore. Thus, we are removing it.

However, the DOMAIN linked to the monitor survives in the database after the monitor is stopped, and is not removed until `clear_monitor` is called. Also, `init_monitor` over the same domain ID will fail because the DOMAIN is still there

But maybe there is a reason beyond the init/stop/restart to keep the inactive monitors in here, glad to discuss if you find it necessary.

Reply:
Ok, I didn't know that `monitors_by_entity_` only have active monitors. I can't find a reason to keep the inactive monitors there, I understand that this attribute exists for access easily to active monitors.
Maybe will be fine to add a commentary to the attribute explaining that only contains active monitors?